dags/veda_data_pipeline/utils/collection_generation.py

```diff
@@ -0,0 +1,135 @@
+from typing import Any, Dict
+
+import fsspec
+import xarray as xr
+import xstac
+from veda_data_pipeline.utils.schemas import SpatioTemporalExtent
+
+
+class GenerateCollection:
+    common_fields = [
+        "title",
+        "description",
+        "license",
+        "links",
+        "time_density",
+        "is_periodic",
+    ]
+    common = {
+        "links": [],
+        "extent": {
+            "spatial": {"bbox": [[-180, -90, 180, 90]]},
+            "temporal": {"interval": [[None, None]]},
+        },
+        "type": "Collection",
+        "stac_version": "1.0.0",
+    }
+
+    def get_template(self, dataset: Dict[str, Any]) -> dict:
+        collection_dict = {
+            "id": dataset["collection"],
+            **GenerateCollection.common,
+            **{
+                key: dataset[key]
+                for key in GenerateCollection.common_fields
+                if key in dataset.keys()
+            },
+        }
+
+        # Default REQUIRED fields
+        if not collection_dict.get("title"):
+            collection_dict["description"] = dataset["collection"]
```

Comment:
should this be `collection_dict["title"] = dataset["collection"]`?

Reply:
`title` isn't a required field but `description` is

Reply:
Ah, that makes sense then. Should it be `if not collection_dict.get("description")` instead of title?

Reply:
mmhmm mmhmm, :nod: